app/controllers/ideas/votes_controller.rb

```diff
@@ -18,16 +18,53 @@ def create
       end
 
       vote = @idea.voted_by current_user
+
       vote.participate = params[:participate]
       vote.save
 
       redirect_to path,
                   flash: {
-                    page_alert: "Your support of <strong>#{@idea.name}</strong> is appreciated!",
+                    page_alert: "Your support of #{@idea.name} is appreciated!",
                     page_alert_type: 'success'
                   }
 
     end
 
+    def update
+
+      path = params[:return_to] || idea_path(@idea)
+      vote = @idea.voted_by current_user
+
+      if vote.participate && params[:participate]=="false"
+        page_alert_msg = "You have withrawn your participation from #{@idea.name}"
+      elsif ! vote.participate && params[:participate]=="true"
+        page_alert_msg = "Thank you for participating in #{@idea.name}!"
+      end
+
+      vote.participate = params[:participate]
+      vote.save
+
```

Comment:
Lines 38-45 could be put in the model to clean up the controller.

page_alert_message = vote.save_participation(@idea, params[:participate])